src/connector/src/parser/parquet_parser.rs

```diff
@@ -188,3 +195,81 @@ impl ParquetParser {
         Ok(data_chunk.into())
     }
 }
+
+/// Retrieves the total number of rows in the specified Parquet file.
+///
+/// This function constructs an `OpenDAL` operator using the information
+/// from the provided `source_desc`. It then accesses the metadata of the
+/// Parquet file to determine and return the total row count.
+///
+/// # Arguments
+///
+/// * `file_name` - The parquet file name.
+/// * `source_desc` - A struct or type containing the necessary information
+///                   to construct the `OpenDAL` operator.
+///
+/// # Returns
+///
+/// Returns the total number of rows in the Parquet file as a `usize`.
+pub async fn get_total_row_nums_for_parquet_file(
+    parquet_file_name: &str,
+    source_desc: SourceDesc,
+) -> ConnectorResult<usize> {
+    let total_row_num = match source_desc.source.config {
+        ConnectorProperties::Gcs(prop) => {
+            let connector: OpendalEnumerator<OpendalGcs> =
+                OpendalEnumerator::new_gcs_source(*prop)?;
+            let reader = connector
+                .op
+                .reader_with(parquet_file_name)
+                .into_future()
+                .await?
+                .into_futures_async_read(..)
+                .await?
+                .compat();
+
+            ParquetRecordBatchStreamBuilder::new(reader)
+                .await?
+                .metadata()
+                .file_metadata()
+                .num_rows()
```

Comment:
You can skip creating ParquetRecordBatchStreamBuilder and use the reader.get_metadata() to directly get the num_rows instead:
```rust
                reader.get_metadata().await?.file_metadata().num_rows();

```